File: backend/src/main/java/github/lms/lemuel/xr/auth/application/GetCurrentUserUseCase.java
package github.lms.lemuel.xr.auth.application;

import github.lms.lemuel.xr.auth.adapter.out.persistence.UserJpaEntity;
import github.lms.lemuel.xr.auth.adapter.out.persistence.UserRepository;
import github.lms.lemuel.xr.common.AppException;
import github.lms.lemuel.xr.common.ErrorCode;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class GetCurrentUserUseCase {

    private final UserRepository users;

    @Transactional(readOnly = true)
    public UserJpaEntity execute(UUID userId) {
        return users.findById(userId)
                .orElseThrow(() -> new AppException(ErrorCode.E_AUTH_REQUIRED, "User not found"));
    }
}

File: backend/src/main/kotlin/github/lms/lemuel/xr/auth/application/GetCurrentUserUseCase.kt
package github.lms.lemuel.xr.auth.application

import github.lms.lemuel.xr.auth.application.port.out.UserPort
import github.lms.lemuel.xr.auth.domain.User
import github.lms.lemuel.xr.common.AppException
import github.lms.lemuel.xr.common.ErrorCode
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Service
class GetCurrentUserUseCase(
    private val users: UserPort,
) {

    @Transactional(readOnly = true)
    fun execute(userId: UUID): User =
        users.findById(userId)
            .orElseThrow { AppException(ErrorCode.E_AUTH_REQUIRED, "User not found") }
}
